### PR TITLE
Fix moveaxis bug

### PR DESCRIPTION
**Pull request by fukatani**
Embarrasingly, my cupy.moveaxis behavior is different from numpy if destination is not ascending order.
I fixed it and added tests.

### CODE REVIEW

**Comment by fukatani** [comment]
We can reproduce bug by follwing script.
```
import numpy
import cupy

a = numpy.ones((4,2,3))
# (3, 4, 2)
print(numpy.moveaxis(a, (0, 2), (1, 0)).shape)

a = cupy.ones((4,2,3))
# (3, 2, 4)
print(cupy.moveaxis(a, (0, 2), (1, 0)).shape)
```

**Comment by fukatani** [comment]
I found one more bug in moveaxis.

```
a = cupy.ones((4,2,3))
# segmentation error
cupy.moveaxis(a, [0, 1], [2, 2])
```

If anyone want to me to split PR, I made another PR.

**Comment by fukatani** [comment]
Thanks! I addressed your comments.

**Comment by unnonouno** [comment]
jenkins test this please

File: cupy/core/core.pyx [hunk 1]
```diff
@@ -2137,9 +2142,9 @@ cpdef ndarray moveaxis(ndarray a, source, destination):
         if not _has_element(src, n):
             order.push_back(n)
 
-    for i in range(len(src)):
-        n = <Py_ssize_t>i
-        order.insert(order.begin() + dest[n], src[n])
+    cdef Py_ssize_t d, s
+    for d, s in sorted(six.moves.zip(dest, src)):
```
Comment (unnonouno) [inline cupy/core/core.pyx]: Sorry, my colleague says we need to use `six` in cython. Please use `zip`.

Comment (fukatani) [reply cupy/core/core.pyx]: No problem! I fixed it.

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**